### ISSUE TITLE
one signal app crash assign user id

### ISSUE THREAD

**Issue by omar-bear** [OneSignal/react-native-onesignal#1313]
**Description:**
A big number of users app crash on their phone because of one signal but we cant reproduce the error it happens suddenly
![image](https://user-images.githubusercontent.com/80390318/140518165-5c817d3d-d1d0-4abd-8078-2cab935fa271.png)
**it happens 359 times for 280 users**

**Environment**
`"react-native-onesignal": "4.3.0",`
sdk installed with `yarn add react-native-onesignal`

**Steps to Reproduce Issue:**
we cant really reproduce the error but it happen for a important number of users

**Anything else:**
stack trace from google play console 
```
java.lang.RuntimeException
```
more information
```java
java.lang.RuntimeException: 
  at com.facebook.react.bridge.CallbackImpl.invoke (CallbackImpl.java:26)
  at com.geektime.rnonesignalandroid.RNOneSignal$7.onSuccess (RNOneSignal.java:462)
  at com.onesignal.OneSignalStateSynchronizer$1$1.run (OneSignalStateSynchronizer.java:291)
  at android.os.Handler.handleCallback (Handler.java:938)
  at android.os.Handler.dispatchMessage (Handler.java:99)
  at android.os.Looper.loop (Looper.java:246)
  at android.app.ActivityThread.main (ActivityThread.java:8595)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:602)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1130)
```
with some search the error happens here
```java
   @ReactMethod
   public void setExternalUserId(final String externalId, final String authHashToken, final Callback callback) {
      OneSignal.setExternalUserId(externalId, authHashToken, new OneSignal.OSExternalUserIdUpdateCompletionHandler() {
         @Override
         public void onSuccess(JSONObject results) {
            Log.i("OneSignal", "Completed setting external user id: " + externalId + "with results: " + results.toString());

            if (callback != null)
               callback.invoke(RNUtils.jsonToWritableMap(results));
         }

         @Override
         public void onFailure(OneSignal.ExternalIdError error) {
            if (callback != null)
               callback.invoke(error.getMessage());
         }
      });
   }
```
Exactly in this line 

```java
if callback != null { callback.invoke(RNUtils.jsonToWritableMap(...)) }
```
So i think maybe it happens when we assign userId
```js
      if (id) {
        OneSignal.setExternalUserId(`${id}`, null, (res) => {
          if (!res) {
            logger.error(
              `An error has occured while setting oneSignal user id: ${id}`
            );
          }
        });
      }
```

**Comment by nan-li** [comment]
Adding some notes:
Seems related to https://github.com/OneSignal/react-native-onesignal/issues/1015
The RuntimeException comes from [here](https://github.com/facebook/react-native/blob/b7b59aee84de9b1ae9eaa4a419efbd62760321cf/ReactAndroid/src/main/java/com/facebook/react/bridge/CallbackImpl.java#L26)

**Comment by nan-li** [comment]
Hi @omar-bear,

The `RuntimeException`  you got:

> "Illegal callback invocation from native module. This callback type only permits a single invocation from native code." 

has been fixed and released in [react-native-onesignal 4.3.4](https://github.com/OneSignal/react-native-onesignal/releases/tag/4.3.4). Please upgrade and reach out out if you are still experiencing the issue.
